src/behaviours/seekSafestFood.ts

```diff
@@ -45,12 +45,21 @@ function isItADeadEnd(
  * @param us - us!
  * @param board - the game board
  */
-function areWeTrapped(pathToSnack: Matrix, us: ISnake, board: IBoard): boolean {
+function itsATrap(pathToSnack: Matrix, us: ISnake, board: IBoard): boolean {
+  // Start from the end of the path
+  const invertPath = [...pathToSnack].reverse();
   // This is us after we hit our fitness goals
-  const futureBody: ICoordinate[] = [];
-  // Map where our body will be when we eat food.
-  for (const coordinate of pathToSnack) {
-    futureBody.push({ x: coordinate[0], y: coordinate[1] });
+  let futureBody: ICoordinate[] = [];
+  // Map where our body will be when we follow the path to the food
+  us.body.forEach((coordinate, i) => {
+    futureBody.push({ x: invertPath[i][0], y: invertPath[i][1] });
```

Comment:
if the pathToSnack is short and our body is long won't this error because there wont be an x value at position i?

Reply:
Yep, good call, resolved in latest commit 👍